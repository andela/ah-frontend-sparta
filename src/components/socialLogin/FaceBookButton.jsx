import React, { Component } from 'react';
import FacebookLogin from 'react-facebook-login';
import loginWithFb from '../../actions/facebookActions';
import store from '../../store';
import './FaceBook.css';

class FaceBookButton extends Component {
   responseFacebook = (response) => {
     const FbToken = response.accessToken;
     store.dispatch(loginWithFb(FbToken));
   };

   render() {
     return (
       <div className="facebookbutton">
         <FacebookLogin
           appId={process.env.appId}
           fields="name,email,picture"
           callback={this.responseFacebook}
           textButton={false}
           icon={<i className="fab fa-facebook" />}
         />
       </div>
     );
   }
}

export default FaceBookButton;
