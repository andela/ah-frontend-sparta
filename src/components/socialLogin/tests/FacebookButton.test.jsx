import React from 'react';
import { shallow } from 'enzyme';
import FaceBookButton from '../faceBookButton';


describe('Facebook Button Tests', () => {
  it('should not regress', () => {
    const componentWrapper = shallow(<FaceBookButton />);
    const response = {
      accessToken: 'testToken',
    };
    componentWrapper.instance().responseFacebook(response);
  });

  it('snapshot test', () => {
    const componentWrapper = shallow(<FaceBookButton />);
    expect(componentWrapper).toMatchSnapshot();
  });
});
