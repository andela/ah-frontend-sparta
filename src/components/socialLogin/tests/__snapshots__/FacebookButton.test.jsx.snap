// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facebook Button Tests snapshot test 1`] = `ShallowWrapper {}`;
